crates/tx/src/types.rs

```diff
@@ -1855,6 +809,17 @@ impl<'tx> Tx {
     }
 }
 
+/// A key to select an inner tx and its sections
+#[derive(Debug, Clone)]
+pub enum InnerTxRef<'a> {
+    /// A direct ref to tx commitments
+    Commitment(&'a TxCommitments),
```

Comment:
If this was an owned value instead of a ref we could use `InnerTx` inside both `BatchedTx` and `BatchedTxRef` in place of `TxCommitments` but this would imply many more changes and those two struct are only used within the protocol, don't know if you have a strong opinion on this